examples/vision/yolov8.py

```diff
@@ -0,0 +1,628 @@
+boxes. These layers intelligently adjust the bounding box coordinates as the image is
+transformed, ensuring that the bounding boxes remain accurate and aligned with the
+augmented images.
+
+By leveraging KerasCV's capabilities, developers can conveniently integrate bounding
+box-friendly data augmentation into their object detection pipelines. By performing
+on-the-fly augmentation within a tf.data pipeline, the process becomes seamless and
+efficient, enabling better training and more accurate object detection results.
+"""
+
+augmenter = keras.Sequential(
+    layers=[
+        keras_cv.layers.RandomFlip(mode="horizontal", bounding_box_format="xyxy"),
+        keras_cv.layers.RandomTranslation(
+            height_factor=0.2, width_factor=0.2, bounding_box_format="xyxy"
+        ),
+        keras_cv.layers.RandomShear(
+            x_factor=0.2, y_factor=0.2, bounding_box_format="xyxy"
+        ),
+        keras_cv.layers.JitteredResize(
+            target_size=(640, 640), scale_factor=(0.75, 1.3), bounding_box_format="xyxy"
+        ),
+    ]
+)
+
+"""
+## Creating Training Dataset 
+"""
+
+train_ds = train_data.map(load_dataset, num_parallel_calls=tf.data.AUTOTUNE)
+train_ds = train_ds.shuffle(BATCH_SIZE * 4)
+train_ds = train_ds.ragged_batch(BATCH_SIZE, drop_remainder=True)
+train_ds = train_ds.map(augmenter, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+## Creating Validation Dataset
+"""
+
+resizing = keras_cv.layers.Resizing(
+    640, 640, bounding_box_format="xyxy", pad_to_aspect_ratio=True
+)
+
+val_ds = val_data.map(load_dataset, num_parallel_calls=tf.data.AUTOTUNE)
+val_ds = val_ds.shuffle(BATCH_SIZE * 4)
+val_ds = val_ds.ragged_batch(BATCH_SIZE, drop_remainder=True)
+val_ds = val_ds.map(resizing, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+## Visualization
+"""
+
+
+def visualize_dataset(inputs, value_range, rows, cols, bounding_box_format):
+    inputs = next(iter(inputs.take(1)))
+    images, bounding_boxes = inputs["images"], inputs["bounding_boxes"]
+    visualization.plot_bounding_box_gallery(
+        images,
+        value_range=value_range,
+        rows=rows,
+        cols=cols,
+        y_true=bounding_boxes,
+        scale=5,
+        font_scale=0.7,
+        bounding_box_format=bounding_box_format,
+        class_mapping=class_mapping,
+    )
+
+
+visualize_dataset(
+    train_ds, bounding_box_format="xyxy", value_range=(0, 255), rows=2, cols=2
+)
+
+visualize_dataset(
+    val_ds, bounding_box_format="xyxy", value_range=(0, 255), rows=2, cols=2
+)
+
+"""
+We need to extract the inputs from the preprocessing dictionary and get them ready to be
+fed into the model. If we're using TPU, we need to make sure that the bounding box
+Tensors are Dense rather than Ragged. If we're training on GPU, we don't need to make
+this change and can skip the `bounding_box.to_dense()` call.
+"""
+
+
+def dict_to_tuple(inputs):
+    return inputs["images"], bounding_box.to_dense(
+        inputs["bounding_boxes"], max_boxes=32
+    )
```

Comment:
https://github.com/keras-team/keras-cv/issues/1832

Reply:
Yeah, that's why we use `to_dense`